### ISSUE TITLE
``ZipParameters.setIncludeRootFolder(false)`` breaks zip generation

### ISSUE THREAD

**Issue by NichtStudioCode** [srikanth-lingala/zip4j#439]
In zip4j 2.11.0, when ``ZipParameters#includeRootFolder`` is set to ``false`` and a folder is added to a ZipFile using ``ZipFile#addFolder``, no files are added at all.
This was not the case in zip4j 2.10.0

**Comment by srikanth-lingala** [comment]
It works fine for me. I added a folder to a zip and set the `includeRootFolder` to `false`, and I got the expected output.

```
ZipFile zipFile = new ZipFile(BASE_DIR + "Test.zip");
ZipParameters zipParameters = new ZipParameters();
zipParameters.setIncludeRootFolder(false);
zipFile.addFolder(new File(BASE_DIR, "root_folder"), zipParameters);

zipFile = new ZipFile(BASE_DIR + "Test.zip");
zipFile.extractAll(BASE_DIR + "SOME_OUTPUT");
```

I attached the folder I tried with as well. In the attached example the root folder `root_folder` has a text file, and a sub-directory again containing another text file. The generated zip file has all this content, without the root folder `root_folder` in it. I attached the folder from my test case, and also the generated zip file.

[root_folder.zip](https://github.com/srikanth-lingala/zip4j/files/8962635/root_folder.zip)
[generated_zip_file.zip](https://github.com/srikanth-lingala/zip4j/files/8962636/generated_zip_file.zip)
